### PR TITLE
⚡️ Opti conditions

### PR DESCRIPTION
**Pull request by makcandrov**
`percentageMul`: 413 => 409
`percentageDiv`: 403 => 402

`rayMul`: 479 => 475
`rayDiv`: 469 => 468
`wadMul`: 435 => 431
`wadDiv` 513 => 512

### CODE REVIEW

**Comment by AtomicAzzaz** [comment]
Nice one, works also in wadRayMath for 4 gas savings in differents Mul

**Review by Rubilmax — APPROVED** [review]
jpp
it gets unreadable but it works

**Comment by makcandrov** [comment]
> Nice one, works also in wadRayMath for 4 gas savings in differents Mul

Updated in the last commit :)

File: src/math/WadRayMath.sol [hunk 1]
```diff
@@ -30,7 +30,7 @@ library WadRayMath {
         // <=> x * y > type(uint256).max - HALF_WAD
         // <=> x > (type(uint256).max - HALF_WAD) / y
         assembly {
-            if and(gt(y, 0), gt(x, div(MAX_UINT256_MINUS_HALF_WAD, y))) {
+            if mul(y, gt(x, div(MAX_UINT256_MINUS_HALF_WAD, y))) {
```
Comment (QGarchery) [inline src/math/WadRayMath.sol:33]: To be precise we can add the condition `y != 0` in the last line in the comment above

**Review by QGarchery — APPROVED** [review]
Ok this should work yes. Although I don't understand why the div opti saves some gas: replace one OR (3) with both a MUL (5) and a ISZERO (3) ...

File: src/math/WadRayMath.sol (hunk 1 above)
Comment (makcandrov) [reply src/math/WadRayMath.sol:33]: `y` is already declared positive L28

```solidity
// Let y > 0
```

Comment (QGarchery) [reply src/math/WadRayMath.sol:33]: True, I missed that. But in the case of multiplication it's not clear why we would assume that at the start. In contrast, the last equivalence makes it clear why we are checking this condition: we are doing a division by `y`

**Review by MerlinEgalite — APPROVED** [review]
LGTM

**Review by MathisGD — COMMENTED** [review]
LGTM

**Pull request merged**